File: EASSET.Web/Modules/ServerTypes/MasterData/SubUnitForm.ts
import { PrefixedContext, initFormType } from "@serenity-is/corelib";
import { ServiceLookupEditor, StringEditor, TextAreaEditor } from "serenity.corelib";

export interface SubUnitForm {
    UnitId: ServiceLookupEditor;
    SubUnitKode: StringEditor;
    SubUnitNama: StringEditor;
    Npwp: StringEditor;
    Alamat: TextAreaEditor;
}

export class SubUnitForm extends PrefixedContext {
    static readonly formKey = 'MasterData.SubUnit';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!SubUnitForm.init)  {
            SubUnitForm.init = true;

            var w0 = ServiceLookupEditor;
            var w1 = StringEditor;
            var w2 = TextAreaEditor;

            initFormType(SubUnitForm, [
                'UnitId', w0,
                'SubUnitKode', w1,
                'SubUnitNama', w1,
                'Npwp', w1,
                'Alamat', w2
            ]);
        }
    }
}

File: EASSET.Web/Modules/MasterData/SubUnit/SubUnitForm.cs
using Serenity.ComponentModel;
using System;

namespace EASSET.MasterData.Forms;

[FormScript("MasterData.SubUnit")]
[BasedOnRow(typeof(SubUnitRow), CheckNames = true)]
public class SubUnitForm
{
    public string UnitId { get; set; }
    public string SubUnitKode { get; set; }
    public string SubUnitNama { get; set; }
    public string Npwp { get; set; }
    //public string ThAng { get; set; }
    //public string Op { get; set; }
    //public DateTime Lu { get; set; }
    //public string Pc { get; set; }
    //public bool Dlt { get; set; }
    [TextAreaEditor(Rows =5, Cols =10)]
    public string Alamat { get; set; }
 
}